packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -535,6 +535,12 @@ export default class ScriptTransformer {
   }
 }
 
+export const buildDefaultResolverRequire = (config: Config.ProjectConfig) => <
```

Comment:
do we need this to be a factory? Is it used more than once anywhere? If not, just a

```ts
export const requireAndTranspileModule = <T>(
  config: Config.ProjectConfig,
  resolverPath: string,
) => new ScriptTransformer(config).requireAndTranspileModule<T>(resolverPath);
```

should work, no?

Reply:
Ideally, but it needs to be a factory here: https://github.com/facebook/jest/pull/8829/files#diff-79e46c9e92027daadc4fc85b7b71ae2dR82-R86

Reply:
fair enough 🙂 Could we avoid creating the transformer more than once then?

```ts
export function buildDefaultResolverRequire(config: Config.ProjectConfig) {
  const transformer = new ScriptTransformer(config);

  return function requireAndTranspileModule<T>(resolverPath: string) {
    return transformer.requireAndTranspileModule<T>(resolverPath);
  };
}
```

(named functions for better stack traces)

---

I'm also not really happy with the name. It contains nothing about the fact modules required through it is transformed (although it's from `@jest/transform` package, so maybe fine?). what about e.g. `createTranspilingRequire` or something like that?

I also wonder if we should automatically use `interopRequireDefault` inside `requireAndTranspileModule`, but that's a separate discussion from this helper

Reply:
@G-Rath ping

Reply:
@SimenB pong! That much have gotten lost in the mail - I'll have a read over and action that first thing morning :)

Reply:
>I'll have a read over and action that first thing morning :)

👍 

>The question then would be if it should sit on the inner or outer function 🤔

Inner methinks, as it's per import. Feel free to open a separate PR for it 🙂